src/main/java/io/vertx/httpproxy/impl/ParseUtils.java

```diff
@@ -87,17 +80,17 @@ public static Date parseHttpDate(String value) throws Exception {
           String s = value.substring(0, sep);
           if (parseWkday(s) != null) {
             // rfc1123-date
-            return RFC_1123_DATE_TIME().parse(value);
+            return Instant.from(DateTimeFormatter.RFC_1123_DATE_TIME.parse(value));
           } else if (parseWeekday(s) != null) {
             // rfc850-date
-            return RFC_850_DATE_TIME().parse(value);
+            return Instant.from(RFC_850_DATE_TIME.parse(value));
           }
           return null;
         }  else if (c == ' ') {
           String s = value.substring(0, sep);
           if (parseWkday(s) != null) {
             // asctime-date
-            return ASC_TIME().parse(value);
+            return Instant.from(ASC_TIME.parse(value));
           }
           return null;
```

Comment:
I think we could simplify this method to:

```java
  public static Instant parseHttpDate(String value) throws Exception {
    int pos = value.indexOf(',');
    if (pos == 3) { // e.g. Sun, 06 Nov 1994 08:49:37 GMT
      return RFC_1123_DATE_TIME.parse(value, Instant::from);
    }
    if (pos == -1) { // e.g. Sun Nov  6 08:49:37 1994
      return ANSI_C_ASCTIME.parse(value, Instant::from);
    }
    return RFC_850_DATE_TIME.parse(value, Instant::from); // e.g. Sunday, 06-Nov-94 08:49:37 GMT
  }
```

See https://www.rfc-editor.org/rfc/rfc9110.html#name-date-time-formats

Reply:
And then you can get rid of the two methods below